src/pwa-buildpack/project-setup/create-configuration-files/index.md

```diff
@@ -0,0 +1,268 @@
+
+The locations specified in the snippet are the standard locations of source code, static assets, and build output in a [Peregrine] app.
+
+### Export the Webpack configuration object
+
+Append the following content to webpack.config.js to export the configuration object for Webpack:
+
+``` javascript
+module.exports = async function(env) {
+    const config = {
+        mode: env.mode, // passed on the command line via the '--env' flag
+        context: __dirname, // Node global for the running script's directory
+        entry: {
+            client: path.resolve(themePaths.src, 'index.js')
+        },
+        output: {
+            path: themePaths.output,
+            publicPath: process.env.MAGENTO_BACKEND_PUBLIC_PATH,
+            filename: '[name].js',
+            chunkFilename: '[name].js'
+        },
+        module: {
+            rules: [
+                {
+                    include: [themePaths.src],
+                    test: /\.js$/,
+                    use: [
+                        {
+                            loader: 'babel-loader',
+                            options: { cacheDirectory: true }
+                        }
+                    ]
+                },
+                {
+                     test: /\.css$/,
+                     use: [
+                         'style-loader',
+                         {
+                             loader: 'css-loader',
+                             options: {
+                                 importLoaders: 1
+                             }
+                         }
+                     ]
+                 }
+             ]
+         },
+         resolve: await MagentoResolver.configure({
+             paths: {
+                 root: __dirname
+             }
+         }),
+         plugins: [
+             new MagentoRootComponentsPlugin(),
+             new webpack.NoEmitOnErrorsPlugin(),
+             new webpack.EnvironmentPlugin({
+                 NODE_ENV: env.mode,
+                 SERVICE_WORKER_FILE_NAME: 'sw.js'
+             })
+         ]
+
+    };
+
+    return config;
+}
+```
+
+This configuration sets up Webpack for your development environment.
+Some important things to note in this configuration:
+
+* How it configures the `MagentoResolver` service
+* The inclusion of the `MagentoRootComponent` as a plugin
+* The use of `webpack.EnvironmentPlugin` to pass environment variables
+
+### Add development mode configuration
+
+Add the following development mode configuration before returning the `config` object:
+
+``` javascript
+if (env.mode === "development") {
+    config.devServer = await PWADevServer.configure({
+        publicPath: process.env.MAGENTO_BACKEND_PUBLIC_PATH,
+        backendDomain: process.env.MAGENTO_BACKEND_DOMAIN,
+        serviceWorkerFileName: process.env.SERVICE_WORKER_FILE_NAME,
+        paths: themePaths,
+        id: 'magento-my-theme'
```

Comment:
Update this to `id: path.basename(__dirname) // Defaults to theme directory name`